olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -288,6 +326,91 @@ model::Partition PartitionsRepository::PartitionFromSubQuad(
+  if (fetch_option != OnlineOnly && fetch_option != CacheWithUpdate) {
+    repository::PartitionsCacheRepository repository(
+        catalog, settings.cache, settings.default_cache_expiration);
+
+    // handle tiles from level < kAggregateQuadTreeDepth
+    auto max_depth =
+        std::min<int>(int(tile_key.Level()), kAggregateQuadTreeDepth);
+    QuadTreeIndex cached_tree;
+    // search in the cache for any quad tree covering the tile
+    for (auto i = max_depth; i >= 0; --i) {
+      const auto& root_tile_key = tile_key.ChangedLevelBy(-i);
+      cached_tree = repository.Get(layer, root_tile_key,
+                                   kAggregateQuadTreeDepth, version);
+      if (!cached_tree.IsNull()) {
+        OLP_SDK_LOG_DEBUG_F(kLogTag, "cache QuadTree (%s, %" PRId32 ") found!",
+                            root_tile_key.ToHereTile().c_str(),
+                            kAggregateQuadTreeDepth);
+
+        return FindPartition(cached_tree, root_tile_key, request, aggregated);
+      }
+    }
+
+    if (cached_tree.IsNull() && fetch_option == CacheOnly) {
+      OLP_SDK_LOG_INFO_F(kLogTag, "cache QuadTree for Tile(%s) not found!",
+                         tile_key.ToHereTile().c_str());
+      return client::ApiError(
+          client::ErrorCode::NotFound,
+          "Cache only resource not found in cache (QuadTree).");
+    }
+  }
+
+  // quad tree data not found in the cache
+  auto query_api = ApiClientLookup::LookupApi(
+      catalog, context, "query", "v1", request.GetFetchOption(), settings);
+
+  if (!query_api.IsSuccessful()) {
+    OLP_SDK_LOG_ERROR(kLogTag,
+                      "QueryPartitionForVersionedTile: LookupApi failed.");
+    return query_api.GetError();
+  }
+
+  const auto& root_tile_key = tile_key.ChangedLevelBy(-kAggregateQuadTreeDepth);
+  const auto root_tile_here = root_tile_key.ToHereTile();
+  auto quadtree_response = QueryApi::QuadTreeIndex(
+      query_api.GetResult(), layer, root_tile_here, version,
+      kAggregateQuadTreeDepth, boost::none, request.GetBillingTag(), context);
+
+  if (quadtree_response.status != olp::http::HttpStatusCode::OK) {
+    OLP_SDK_LOG_ERROR_F(kLogTag,
+                        "QuadTreeIndex failed (%s, %" PRId64 ", %" PRId32 ")",
+                        root_tile_here.c_str(), version.get_value_or(-1),
+                        kAggregateQuadTreeDepth);
+    return client::ApiError(quadtree_response.status,
+                            quadtree_response.response.str());
+  }
+
+  QuadTreeIndex tree(root_tile_key, kAggregateQuadTreeDepth,
+                     quadtree_response.response);
+  if (tree.IsNull()) {
+    OLP_SDK_LOG_ERROR_F(kLogTag,
+                        "QuadTreeIndex failed (%s, %" PRId64 ", %" PRId32 ")",
+                        root_tile_here.c_str(), version.get_value_or(-1),
+                        kAggregateQuadTreeDepth);
+    return {
+        {client::ErrorCode::Unknown, "Failed to parse QuadTreeIndex json."}};
```

Comment:
```suggestion
        {client::ErrorCode::Unknown, "Failed to parse QuadTreeIndex json"}};
```
Formatting seems wrong!